File: 1302-deepest-leaves-sum/1302-deepest-leaves-sum.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int deepestLeavesSum(TreeNode root) {
        //If tree Doesnt exist
        if(root == null) return 0;
        int sum = 0;
        LinkedList<TreeNode> que = new LinkedList<>();
        que.addLast(root);
        //Start Level Order Traversal
        while(que.size()!=0) {
            int size = que.size();
            //initialize sum = 0 for each new levl
            sum = 0;
            while(size-- >0) {
                
                TreeNode removeN = que.removeFirst();
                //If left and right exists add in que
                if(removeN.left != null) que.addLast(removeN.left);
                
                if(removeN.right != null) que.addLast(removeN.right);
                //add all the nodes sum of same level in sum.
                sum += removeN.val;
            }
        }
        return sum;
    }
}

File: 1302-deepest-leaves-sum/1302-deepest-leaves-sum.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int deepestLeavesSum(TreeNode* root) {
        //Base condn
        if(root == nullptr) return 0;
        // Create Que
        queue<TreeNode*> que;
        que.push(root);
        int sum = 0;
        while(que.size()!=0) {
            int size = que.size();
            sum = 0;
            //maintain sum as 0 for each new level
            while(size-- > 0) {
                
                TreeNode* removeN = que.front(); que.pop();
                //if left and right child exists add in que.
                if(removeN->left != nullptr) que.push(removeN->left);
                
                if(removeN->right != nullptr) que.push(removeN->right);
                // sum of last level will get updated in latest sum.
                sum += removeN->val;
            }
        }
        
        return sum;
    }
};